lib/features/home/home_screen.dart

```diff
@@ -16,103 +19,64 @@ class IDHome extends StatefulWidget {
 }
 
 class _IDHomeState extends State<IDHome> with WidgetsBindingObserver {
+
+  // @override
+  // void initState() {
+  //   // TODO: implement initState
+  //   super.initState();
+  // }
+
   @override
-  Widget build(BuildContext context) => Scaffold(
-        body: BlocBuilder<LocationBloc, LocationState>(
-          builder: (BuildContext context, LocationState state) {
-            if (state is LocationReadyState) {
-              return Stack(
-                children: <Widget>[
-                  GoogleMapView(
-                    currentLocation: state.currentLocation,
-                  ),
-                  SafeArea(
-                      child: Align(
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: BlocConsumer<LocationBloc, LocationState>(
+        listener: (_, LocationState state) {
+          if (state is LocationPermissionDeniedState || state is LocationPermissionDeniedForeverState) {
+            context.showIDBottomSheet(
+              iconPath: Assets.images.idLocationImg,
+              title: S.of(context).locationTurnOnSuggestionTitle,
+              content: S.of(context).locationTurnOnSuggestionDescription,
+              primaryButtonText: S.of(context).ctaGoToSettings,
+              textButtonText: S.of(context).ctaLocationTurnOnContinueDefaultLocation,
+              onPrimaryButtonSelected: () => LocationService().requestOpenAppSettings(),
```

Comment:
when state is LocationPermissionDeniedState, primary button should send event ShowDialogRequestPermissionEvent()

when state is LocationPermissionDeniedForeverState, primary button should LocationService().requestOpenAppSettings

Reply:
@suesitran i assumed that you mean 
`state is LocationPermissionDeniedState` -> Fire event ShowDialogRequestPermissionEvent
`state is LocationPermissionDeniedForeverState` -> Show bottom sheet.

is that right ? 

I have one more question about behaviour:
- If user don't allow permission at first time. We immediately show Bottom Sheet or we only show it the next time user open app ?

Reply:
No, in both the states, we show bottom popup, but the behaviours in primary button onPressed are different for each state, and the display text for primary button are different as well. Please check the old code for more details.

Reply:
OLD VERSION 
- `LocationPermissionDeniedState`
![image](https://user-images.githubusercontent.com/41529808/206511078-5a11a7c9-35f4-4ea2-a32d-97d8ede636c6.png)
- `LocationPermissionDeniedForeverState`
![image](https://user-images.githubusercontent.com/41529808/206511359-cc0963bb-9dac-4100-bfe2-708ebaf7cb94.png)


NEW VERSION
- Design : 
![image](https://user-images.githubusercontent.com/41529808/206513950-ae5d76bf-c847-472b-81df-9ced2e0b350b.png)

- The content of new version is different from the old version above. So it gonna be really weird if we keep the content (title, description) for both Denied vs DeniedForever state.
- As i understood. DeniedForever mean that user have denied permission before, and they have to manually  enable it again in device settings. 
- So i changed to this:
- `LocationPermissionDeniedState`
![image](https://user-images.githubusercontent.com/41529808/206606014-7cd33660-1a0c-4747-8641-ce06d2ab3dbf.png)
- `LocationPermissionDeniedForeverState`
![image](https://user-images.githubusercontent.com/41529808/206606129-99f466fd-1119-4a24-95ad-d72a389cb972.png)

Reply:
yes, that's correct implementation on the UI now, how about the interaction? 

when `LocationPermissionDeniedState`, and user taps on 'Allow', we need to show native permission request dialog

and when state is `LocationPermissionDeniedForeverState`, when user taps on 'Go to Settings', then we'll open app setting